src/a2m-v2.h

```diff
@@ -0,0 +1,850 @@
+/*
+ * Adplug - Replayer for many OPL2/OPL3 audio file formats.
+ * Copyright (C) 1999 - 2008 Simon Peter, <dn.tlp@gmx.net>, et al.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * a2m-v2.cpp - Adlib Tracker II Player by Dmitry Smagin <dmitry.s.smagin@gmail.com>
+ *              Originally by Stanislav Baranec <subz3ro.altair@gmail.com>
+ *
+ * NOTES:
+ * This player loads a2m and a2t modules versions 1 - 14.
+ * The code is adapted directly from FreePascal sources of the Adlib Tracker II
+ *
+ * REFERENCES:
+ * https://github.com/ijsf/at2
+ * http://www.adlibtracker.net/
+ *
+ */
+
+#ifndef H_ADPLUG_A2MPLAYER
+#define H_ADPLUG_A2MPLAYER
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <string.h>
+#include <assert.h>
+#include <math.h>
+
+#include "player.h"
+#include "depack.h"
+#include "sixpack.h"
+#include "unlzh.h"
+#include "unlzss.h"
+#include "unlzw.h"
+
+#ifndef C_ASSERT
+#define C_ASSERT(e) typedef char __C_ASSERT__[(e) ? 1 : -1]
```

Comment:
As this is C++, you can just use `static_assert`. It results in nicer error messages.

Reply:
Yep, but static_assert is supported starting from c++11, and we aim to support GCC 4.8 at least. However, we can make the usage of static_assert optional.

Reply:
Should work fine with GCC 4.8 though. https://en.cppreference.com/w/cpp/compiler_support/11

Ultimately we can let CI tell us whether it's ok, assuming that the minimum compiler version is covered.